api/src/auth/drivers/openid.ts

```diff
@@ -209,6 +238,8 @@ export class OpenIDAuthDriver extends LocalAuthDriver {
 			// user that is about to be updated
 			let emitPayload: Record<string, unknown> = {
 				auth_data: userPayload.auth_data,
+				// Make sure a user's role gets updated if his openid group or role mapping changes
+				role: role,
```

Comment:
Here's a potentially tricky question:

What would one expect to happen when you manually change somebody's role in Directus, but then the OpenID role overrides that again back to what is configured in the env? 🤔 Is there a case where you'd use the env var as a "first grouping" with the intention to later change the role in Directus itself outside of the OpenID mapping?

I _think_ I'd personally expect the OID mapping to be the source of truth, but that's something we should make very clear in the docs if that's the case 🤔

Reply:
> I think I'd personally expect the OID mapping to be the source of truth, but that's something we should make very clear in the docs if that's the case 🤔

I'm on the same page as you on this. Personally I think it's quite odd to manage some users using OIDC and others using the build in user management (apart from the administrator user).

Grafana for example gets around this problem by making OIDC users read-only to Grafanas internal user management system. Is this something that could easily be done in Directus, too?

Reply:
There's no baked in easy way to conditionally makes things feel readonly no 🤔 Something to keep in mind for a future improvement 🙂

Reply:
The way the LDAP implementation handles this is that if the `GROUP_DN` property is defined (indicating that user roles are defined in active directory), their role will always be re-assigned on authentication based on their AD group (falling back to the default role if no group is found). Otherwise, roles are managed through Directus post-authentication. https://docs.directus.io/self-hosted/config-options.html#ldap-ldap

I believe there was a recent overhaul of how roles are managed, but I'm not familiar with it, so I don't know how this all currently works with that.

Reply:
Thanks as always @aidenfoxx! 🙂

Roles work the same, it's just how permissions are attached to the roles that's different now 👍🏻